process/process_linux_test.go

```diff
@@ -59,3 +60,30 @@ func BenchmarkLinuxAllProcessesOnLocalProcFS(b *testing.B) {
 		b.Fatal("non-zero errors for AllProcesses", errCount)
 	}
 }
+
+func TestFillNSPidFromStatus(t *testing.T) {
+	err := os.Setenv("HOST_PROC", "./resources/linux_inner_ns/proc")
+	defer os.Unsetenv("HOST_PROC")
+	assert.Nil(t, err)
+
+	// Process started on the host namespace
+	// NSpid:	6320
+	p1, err := NewProcess(6320)
+	assert.Nil(t, err)
+	p1.fillFromStatus()
+	assert.Equal(t, int32(6320), p1.NSPid)
+
+	// Process spawned from P1 with its own namespace
+	// NSpid:	6321	1
+	p2, err := NewProcess(6321)
+	assert.Nil(t, err)
+	p2.fillFromStatus()
+	assert.Equal(t, int32(1), p2.NSPid)
+
+	// Process spawned from P2 with its own namespace
+	// NSpid:	6322	2	1
```

Comment:
Curious: would a process have multiple namespace pids?

Reply:
This could happen. Every time a process does a clone or fork, it could pass some parameters to create a new namespace PID for its child. So it could happen that a process has multiple nested namespaces. 

I was considering collecting all of them but for the moment, the most interesting ones are the first (from the host point of view) and the last one (view by the process itself).

Do you think that we should collect all of them? One case that I think that it could be interesting is to monitor inter process communication.